lib/rules/require-valid-categories.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const utils = require('../utils');
+
+module.exports = {
+  meta: {
+    docs: {},
+    category: '', // fixme: add valid category
+    schema: {
```

Comment:
My thought is that this rule should be opt-in only and requires options (otherwise what categories would you validate against)? As such, the only option this rule takes is an array of strings containing at least one value. Otherwise, ESLint should report that the rule was configured improperly. Thoughts on this?

Reply:
My only concern here is that this plugin has the `plugin:eslint-plugin/all` config, which currently claims to enable all rules from the plugin. This isn't possible unless all rules can be run without providing an explicit configuration. So my slight preference would be to have some default set of categories (e.g. `Possible Errors`, `Best Practices`, `Stylistic Issues`) if the user doesn't provide any categories on their own.

Reply:
Perhaps all the categories ESLint core uses? It would be nice to be able to use those and combine them with plugin-specific options.